File: ontrack-repository/src/main/java/net/nemerosa/ontrack/repository/ProjectFavouriteRepository.java
package net.nemerosa.ontrack.repository;

public interface ProjectFavouriteRepository {

    /**
     * Is this project a favourite?
     */
    boolean isProjectFavourite(int accountId, int projectId);

    /**
     * Sets a project as favourite (or not)
     */
    void setProjectFavourite(int accountId, int projectId, boolean favourite);

}

File: ontrack-repository/src/main/java/net/nemerosa/ontrack/repository/ProjectFavouriteRepository.kt
package net.nemerosa.ontrack.repository

interface ProjectFavouriteRepository {

    /**
     * Gets the list of favourite projects for a user
     *
     * @param accountId ID of the user
     * @return List of project IDs
     */
    fun getFavouriteProjects(accountId: Int): List<Int>

    /**
     * Is this project a favourite?
     */
    fun isProjectFavourite(accountId: Int, projectId: Int): Boolean

    /**
     * Sets a project as favourite (or not)
     */
    fun setProjectFavourite(accountId: Int, projectId: Int, favourite: Boolean)
}